django/db/models/query.py

```diff
@@ -510,17 +522,48 @@ def bulk_create(self, objs, batch_size=None, ignore_conflicts=False):
                     obj_with_pk._state.adding = False
                     obj_with_pk._state.db = self.db
             if objs_without_pk:
-                fields = [f for f in fields if not isinstance(f, AutoField)]
-                returned_columns = self._batched_insert(
-                    objs_without_pk, fields, batch_size, ignore_conflicts=ignore_conflicts,
-                )
-                if connection.features.can_return_rows_from_bulk_insert and not ignore_conflicts:
-                    assert len(returned_columns) == len(objs_without_pk)
-                for obj_without_pk, results in zip(objs_without_pk, returned_columns):
-                    for result, field in zip(results, opts.db_returning_fields):
-                        setattr(obj_without_pk, field.attname, result)
-                    obj_without_pk._state.adding = False
-                    obj_without_pk._state.db = self.db
+                if next_concrete_models:
+                    for next_concrete_model in next_concrete_models:
+                        # Recursively bulk insert parent objects.
+                        parent_objs = []
+                        for obj_without_pk in objs_without_pk:
+                            parent_obj_fields = []
+                            parent_obj_values = []
+                            for f in next_concrete_model._meta.concrete_fields:
```

Comment:
I think we should use `local_concrete_fields` :thinking: Also, the current solutions doesn't work with recursive parents, e.g.
```python
Pizzeria.objects.bulk_create([
    Pizzeria(name='Vegan Pizza House', rank=33),
])
```
crashes when we add the `Ranked` model:
```python
class Place(models.Model):
    name = models.CharField(max_length=100)

    class Meta:
        abstract = True


class Ranked(models.Model):
    rank = models.IntegerField(null=True)


class Restaurant(Ranked, Place):
    pass


class Pizzeria(Restaurant):
    pass
````
```

Reply:
Hi, I'm going to make an attempt at finishing this PR, but I'm not super familiar with Django innards.  Do you have any suggestions towards resolving this problem?

Reply:
@felixxm ^

Reply:
@dralley The suggestion is already in the comment: _"I think we should use `local_concrete_fields`"_, but it's only a suggestion. I didn't check this.

Reply:
Oh, I thought it was referring to two separate issues.

Alright then, I'll make a new PR and we can proceed from there